### PR TITLE
fix: editor will close when open it at first time & add commits route support

### PR DESCRIPTION
**Pull request by conwnet**
Bug: editor will close when open it at first time
Steps of recurrence:

1. Open https://github1s.com
2. Open any file, the editor will force closed at the first time

This is because we refresh the explorer state when the pageType changed. In fact the refresh action should not be occured when the pageType is become BLOB from TREE, vice versa. Sorry for my mistake.

https://github.com/conwnet/github1s/blob/bbc88fee5307a48128be0753f481c1b21a549b9b/extensions/github1s/src/listeners/router/explorer.ts#L19-L24

Another things is add the `commits` route support for GitHub.

### CODE REVIEW

File: extensions/github1s/src/listeners/router/explorer.ts [hunk 1]
```diff
@@ -10,19 +10,46 @@ import {
...
+			(key) => currentState[key] !== previousState[key]
+		)
+	) {
+		return true;
+	}
+
+	if (
+		currentState.pageType !== previousState.pageType &&
+		// when the pageType transform to each other between TREE and BLOB,
+		// don't refresh the explorer state
```
Comment (conwnet) [inline extensions/github1s/src/listeners/router/explorer.ts:36]: I fixed it here.

File: extensions/github1s/src/listeners/router/explorer.ts [hunk 2]
```diff
@@ -10,19 +10,46 @@ import {
...
+		!sortedEqual(
+			[currentState.pageType, previousState.pageType],
+			[PageType.TREE, PageType.BLOB]
+		)
```
Comment (conwnet) [inline extensions/github1s/src/listeners/router/explorer.ts:40]: I fixed it here.

**Review by Siddhant-K-code — APPROVED** [review]
LGTM!

**Comment by Siddhant-K-code** [comment]
> 2\. Open any file, the editor will force closed at the first time

what's this, I am unable to get it, can you pls explain.

**Review by xcv58 — APPROVED** [review]

**Comment by conwnet** [comment]
> what's this, I am unable to get it, can you pls explain.

because the `pageType` in route `https://github1s.com/conwnet/github1s` is TREE, and when open a new file (for example `package.json`), the route will change to `https://github1s.com/conwnet/github1s/blob/HEAD/package.json`, the `pageType` changed to `BLOB` here.

The code `https://github.com/conwnet/github1s/blob/bbc88fee5307a48128be0753f481c1b21a549b9b/extensions/github1s/src/listeners/router/explorer.ts#L19-L24` said: 'if the pageType changed, we have to refresh the file explorer and close all editors', so that we met this bug

**Pull request merged**